azure-pipelines.yml

```diff
@@ -607,14 +602,21 @@ stages:
           - bash: python -m pip install --upgrade pip==21.3.1
             displayName: 'Update pip'
 
+          - script: |
+              deployment_version=$(cat ./great_expectations/deployment_version)
+              tag=$(basename "$(Build.SourceBranch)")
+              if [[ "$deployment_version" != "$tag" ]]; then
+                printf "There is a discrepancy between the 'deployment_version' file (%s) and the provided tag (%s)" $deployment_version $tag
+                exit 1
+              fi
+            displayName: 'Check for valid tag'
+
           - script: |
               pip install twine
               pip install wheel
-              echo "Creating tag: $(cat ./great_expectations/deployment_version)"
               git config --global user.email "team@greatexpectations.io"
               git config --global user.name "Great Expectations"
-              git tag --annotate $(cat ./great_expectations/deployment_version) -m "$(cat ./great_expectations/deployment_version)" || echo "tag exists, skipping"
```

Comment:
Tag should always be made before we get here so I got rid of the step completely.